packages/salesforcedx-apex-replay-debugger/src/core/logContext.ts

```diff
@@ -99,6 +113,88 @@ export class LogContext {
     );
   }
 
+  public getHeapDumps(): ApexHeapDump[] {
+    return this.apexHeapDumps;
+  }
+
+  public scanLogForHeapDumpLines(): boolean {
+    const heapDumpRegex = RegExp(/\|HEAP_DUMP\|/);
+    for (const line of this.logLines) {
+      if (heapDumpRegex.test(line)) {
+        const splitLine = line.split('|');
```

Comment:
Could you edit the regex so it makes sure you have all these indexes before accessing them?

Reply:
I'm leaving the regex as is. I can check the split number to ensure the line has the minimum number of fields.